code/control/DeployDispatcher.php

```diff
@@ -57,7 +57,16 @@ public function index(\SS_HTTPRequest $request) {
 	 */
 	public function history(SS_HTTPRequest $request) {
 		$data = [];
-		foreach($this->DeployHistory() as $deployment) {
+		$list = $this->DeployHistory();
+		$page = $request->getVar('page') ?: 1;
+		$start = ($page-1) * $list->getPageLength();
```

Comment:
Can you do some sanity checking, like is_numeric() and clamp it between [0, -> numRecords]?